assets/js/base/components/product-list/style.scss

```diff
@@ -416,21 +420,30 @@
 	}
 
 	.wc-block-grid__products .wc-block-grid__product-onsale {
-		position: absolute;
-		right: 4px;
-		top: 4px;
-		display: inline-block;
 		background: $twentytwenty-highlights-color;
 		color: #fff;
 		font-family: $twentytwenty-headings;
 		font-weight: 700;
 		letter-spacing: -0.02em;
 		line-height: 1.2;
 		text-transform: uppercase;
-		z-index: 1;
 		font-size: 1.5rem;
+	}
+
+	// Override style from WC Core that set its position to absolute.
+	// This ruleset can be removed once https://github.com/woocommerce/woocommerce/pull/26516 is released.
+	.wc-block-grid__products .wc-block-grid__product-onsale.wc-block-component__sale-badge {
+		position: static;
+	}
+
+	// These styles are not applied to the All Products atomic block, so it can be positioned normally.
+	.wc-block-grid__products .wc-block-grid__product-onsale:not(.wc-block-component__sale-badge) {
```

Comment:
Looks like we need to specifically target the image overlay sale badge to get it top-right. It's not working for me in Chrome:

<img width="1167" alt="Screen Shot 2020-05-28 at 9 24 42 AM" src="https://user-images.githubusercontent.com/4167300/83073796-25b76980-a0c5-11ea-8f80-ba01938606c4.png">

Reply:
Ah, I missed that. Should be fixed in 940ab6942171cf692a8b32859ccefd633dea8585.

Reply:
Dang - I'm still seeing the issue?!

<img width="1219" alt="Screen Shot 2020-05-29 at 3 14 45 PM" src="https://user-images.githubusercontent.com/4167300/83217166-23393a80-a1bf-11ea-8afd-12ec12142950.png">

Reply:
Ah, my bad, I forgot they could be aligned to the left and to the center. The CSS selector I wrote was only targeting them when they were right-aligned. :facepalm: 

Should be fixed with 20b9ea424830e907026046cb11577ab19d552dbc.